jobs/delorean/jenkinsfiles/branch/ga/Jenkinsfile

```diff
@@ -107,7 +146,52 @@ node {
                 String title = "[WIP] ${productName} update"
                 String body = "Created automatically by jenkins"
                 GHPullRequest pr = ghFindOrCreatePullRequest(gitHub.getRepository("${ghOwner}/${ghRepo}"), "${ghOwner}:${sourceBranch}", targetBranch, title, body)
-                println "Component Update PR = ${pr.getHtmlUrl()}"
+                prUrl = pr.getHtmlUrl()
+                runIntegrationTests = prHasLabel(pr, integrationTestsGHLabel)
+                println "Component Update PR = ${prUrl}, runIntegrationTests = ${runIntegrationTests}"
+            }
+        }
+    }
+
+    stage('Integration Tests') {
+        when(sourceChanges && runIntegrationTests) {
+            withCredentials([usernamePassword(
+                    credentialsId: githubUserPassCredentialsID,
+                    passwordVariable: 'GITHUB_PASSWORD',
+                    usernameVariable: 'GITHUB_USERNAME')]) {
+                final GitHub gitHub = new GitHubBuilder()
+                        .withOAuthToken(env.GITHUB_PASSWORD, env.GITHUB_USERNAME)
+                        .build()
+                ghUpdatePrCommitStatus(gitHub, prUrl, GHCommitState.PENDING, env.BUILD_URL, 'Pending', ghCommitStatusContext)
+            }
```

Comment:
Tried to refactor this, but kept hitting issues with jenkins pipelines and objects that cant be serialized. Ended up with quite a lot of repeated code, but it does at least work.